ocaml/xapi/test_vdi_allowed_operations.ml

```diff
@@ -262,6 +262,41 @@ let test_cbt =
       )
   in
 
+
+  let test_vdi_data_destroy () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+
+    (* change VDI properties so that data_destroy passes *)
+    let pass_data_destroy vdi =
+      let sr = Db.VDI.get_SR ~__context ~self:vdi in
+      Db.SR.set_is_tools_sr ~__context ~self:sr ~value:false;
+      Db.VDI.set_is_a_snapshot ~__context ~self:vdi ~value:true;
+      Db.VDI.set_cbt_enabled ~__context ~self:vdi ~value:true in
```

Comment:
Could we set the managed field of the VDI to true here?
> This operation is only allowed on managed VDIs (VDIs whose
managed field is true)

I know we cannot check for that error here, but it's still better to have it in this function.